WooCommerce/Classes/ViewRelated/Products/Variations/Bulk Update/BulkUpdateViewModel.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+import Yosemite
+import protocol Storage.StorageManagerType
+
+/// View Model logic for the Bulk Variations Update
+final class BulkUpdateViewModel {
+
+    /// Represents possible states for syncing product variations.
+    enum SyncState: Equatable {
+        case syncing
+        case syncedResults
+        case syncingError
+        case notStarted
+    }
+
+    private let storageManager: StorageManagerType
+    private let storesManager: StoresManager
+    private let siteID: Int64
+    private let productID: Int64
+
+    /// The state of synching all variations
+    private(set) var syncState: SyncState {
+        didSet {
+            observeSyncState?(syncState)
+        }
+    }
+
+    /// Called when syncing state changes
```

Comment:
Did you considered exposing a `combine` publisher?

Reply:
Yes and I prefer a publisher since it is more concise(I also like reactive functional code :) ).
I have updated the to use a publisher.

To be honest I only used closures because this was what I had described in the original implementation plan.